### ISSUE TITLE
Runtime typing Literal usage unnecessary check

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#5185]
**Describe the bug**
```python
def _normalize_type(tp: type) -> type:
    origin_tp = get_origin(tp)
    if origin_tp is None:
        return tp

    if origin_tp is typing.Literal:
        return typing_extensions.Literal[get_args(tp)]  # False positive error
  ...
```

**Expected behavior**
No error instead of `Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value`

**VS Code extension or command-line**
pylance and pyright latest. I'm assuming this a long behavior though as rare to do much type introspection.

**Additional context**
Motivation is runtime typing inspection. That code is related to runtime typeguard [checking](https://github.com/agronholm/typeguard/issues/363).

**Comment by erictraut** [comment]
Pyright is working as expected here. `Literal` is a special form in the type system. It is not a type itself in the type system, so a static type checker will not treat it as such. You'll need to work around this in your code if you are using `typing.Literal` as a runtime class.
